dask/bytes/core.py

```diff
@@ -145,64 +149,63 @@ def read_bytes(urlpath, delimiter=None, not_zero=False, blocksize=2**27,
             if not_zero:
                 off[0] = 1
                 length[0] -= 1
             offsets.append(off)
             lengths.append(length)
-        machines = [[None]] * len(offsets)
+        machines = [None] * len(offsets)
+
+    delayed_read = delayed(read_block_from_file)
 
     out = []
     for path, offset, length, machine in zip(paths, offsets, lengths, machines):
-        ukey = fs.ukey(path)
-        keys = ['read-block-%s-%s' %
-                (o, tokenize(path, compression, offset, ukey, kwargs, delimiter))
-                for o in offset]
-        L = [delayed(read_block_from_file)(myopen(path, mode='rb'), o,
-                                           l, delimiter, dask_key_name=key)
-             for (o, key, l) in zip(offset, keys, length)]
-        out.append(L)
-        if machine is not None:  # blocks are in preferred locations
-            if client is None:
-                try:
-                    from distributed.client import default_client
-                    client = default_client()
-                except (ImportError, ValueError):  # no distributed client
-                    client = False
-            if client:
-                restrictions = {key: w for key, w in zip(keys, machine)}
-                client._send_to_scheduler({'op': 'update-graph', 'tasks': {},
-                                           'dependencies': [], 'keys': [],
-                                           'restrictions': restrictions,
-                                           'loose_restrictions': list(restrictions),
-                                           'client': client.id})
-
-    if sample is not True:
-        nbytes = sample
-    else:
-        nbytes = 10000
+        token = tokenize(fs_token, delimiter, path, fs.ukey(path),
+                         compression, offset)
+        keys = ['read-block-%s-%s' % (o, token) for o in offset]
+        out.append([delayed_read(OpenFile(fs, path, compression=compression),
+                                 o, l, delimiter, dask_key_name=key)
+                    for o, key, l in zip(offset, keys, length)])
+        if machine is not None and client is not None:
+            # blocks are in preferred locations
+            restrictions = {key: w for key, w in zip(keys, machine)}
+            client._send_to_scheduler({'op': 'update-graph',
+                                       'tasks': {},
+                                       'dependencies': [],
+                                       'keys': [],
+                                       'restrictions': restrictions,
+                                       'loose_restrictions': keys,
+                                       'client': client.id})
+
     if sample:
-        # myopen = OpenFileCreator(urlpath, compression)
-        with myopen(paths[0], 'rb') as f:
+        with OpenFile(fs, paths[0], compression=compression) as f:
+            nbytes = 10000 if sample is True else sample
```

Comment:
What if `sample is False`?  Is this not a valid input?

Reply:
We are within an `if sample` block - if `sample is False`, the original value is returned (and presumably ignored) rather than loading any data.

Reply:
Correct, the altered logic is equivalent to the previous version.